invisible_cities/reco/icaro_components_test.py

```diff
@@ -0,0 +1,85 @@
+import numpy         as np
+import numpy.testing as npt
+import pandas        as pd
+
+from hypothesis.strategies import integers
+from hypothesis.strategies import floats
+from hypothesis            import given
+from pytest                import mark
+from pytest                import raises
+
+from   sklearn.linear_model import RANSACRegressor
+
+from .. core.testing_utils  import assert_dataframes_equal
+from .. core                import core_functions   as core
+
+from .                      import icaro_components as icarcomp
+
+@mark.parametrize("nsignals", [0, 1, 100, 9999, 10*1000])
+@mark.parametrize("signal"  , icarcomp.type_of_signal)
+def test_select_nS_mask_and_check_right_output(nsignals, signal):
+    nevt = 10*1000
+    data = np.concatenate([np.zeros(nevt- nsignals), np.ones(nsignals)])
+    np.random.shuffle(data)
+    data = pd.DataFrame({'nS1': data, 'nS2': data, 'event': range(nevt)})
+    mask = icarcomp.select_nS_mask_and_check(data, signal)
+    assert np.count_nonzero(mask) == nsignals
+
+
+@mark.parametrize("nsignals", [1, 100, 9999, 10*1000])
+def test_select_nS_mask_and_check_consistency(nsignals):
+    nevt = 10*1000
+    data = np.concatenate([np.zeros(nevt - nsignals), np.ones(nsignals)])
+    np.random.shuffle(data)
+    data    = pd.DataFrame({'nS1': data, 'event': range(nevt)})
+    mask    = icarcomp.select_nS_mask_and_check(data, icarcomp.type_of_signal.nS1)
+    mask_re = icarcomp.select_nS_mask_and_check(data, icarcomp.type_of_signal.nS1, input_mask=mask)
+    npt.assert_equal(mask, mask_re)
```

Comment:
```suggestion
    npt.assert_equal(mask_re, mask)
```
Although it's commutative, we tend to (with exceptions) write the target as the first argument and the expected value as the second. This applies also to `==`  and other comparisons.